commands/build.go

```diff
@@ -108,6 +104,12 @@ func runBuild(dockerCli command.Cli, in buildOptions) (err error) {
 		pull = *in.pull
 	}
 
+	if in.quiet && in.progress != "auto" && in.progress != "quiet" {
+		return errors.Errorf("progress=%s and quiet cannot be used together", in.progress)
```

Comment:
I wonder if this should be an error. Maybe `-q` just overrides. Thinking of scripts where different components add their flags.

Reply:
My thinking was that explicitly asking for "fancy output"/"streamed output" and "no output" was user-error and should be rejected, can a component add a new `--progress` type?.

I allowed the user to pass `--progress=quiet -q` which is (possibly surprisingly) different to passing `--progress=quiet` on its own (the former doesn't print the digest at the end), since that's not contradictory.

Reply:
I thought for a case where `--progress=plain` is used to not detect tty and then another part does not want progress at all.